pkg/destinations/destinations.go

```diff
@@ -56,13 +66,21 @@ func (m *DestinationManager) Destination(databaseID int64) (Destination, error)
 		defer m.mux.Unlock(databaseID)
 
 		var dest Destination
+		var creds Creds
 
 		dest, ok := m.pool[databaseID]
 		if ok {
 			return dest, nil
 		}
 
-		creds, err := m.storage.Database.GetDestinationCredentials(databaseID)
+		jsonCreds, err := m.storage.Vault.GetCredential(strconv.Itoa(int(databaseID)))
```

Comment:
@poundifdef in which format should the credentials be stored on aws secrets manager ?
Also, should this logic be different when reading from different vaults. As the memory vault returns a json string consisting of all destination details, which is here decoded to Credentials type.